Reactions/ReactorBase.cpp

```diff
@@ -96,29 +97,51 @@ ReactorBase::set_sundials_solver_tols(
                      << " tolerances with TypVals rtol = " << relTol
                      << " atolfact = " << absTol << " in PelePhysics \n";
     }
-    for (int i = 0; i < ncells; i++) {
-      const int offset = i * (NUM_SPECIES + 1);
-      for (int k = 0; k < NUM_SPECIES + 1; k++) {
-        ratol[offset + k] = m_typ_vals[k] * absTol;
+#if defined(AMREX_USE_HIP) || defined(AMREX_USE_CUDA)
+    const int nbThreads = 256;
+    const int nbBlocks = std::max(1, neq_tot / nbThreads);
+    auto arr = m_typ_vals.data();
```

Comment:
No need for the `.data()`, the GpuArray can be captured by the lambda directly.

Reply:
great point. fixed